evcouplings/couplings/mean_field.py

```diff
@@ -549,6 +589,175 @@ def to_raw_ec_file(self, couplings_file):
                         "{0:.6f}".format(self.di_scores[i, j])
                     ])) + "\n")
 
+    def adapt_from_plmc_reading(self):
+        """
+        Adaptions that allow to read
+        a mean-field couplings model
+        from file where __read_plmc_v2
+        in CouplingsModel does the
+        heavy lifting.
+
+        This includes:
+        - Manage unused plmc-specific
+        fields as well as the pseudo count
+        field
+        - Modify pair frequencies
+        - Regularize column and pair
+        frequencies (i.e. fill the fields
+        regularized_f_i and regularized_f_ij)
+        """
+        self.decode_unused_fields()
+
+        # set the frequency of a pair (alpha, alpha)
+        # in position i to the respective single-site
+        # frequency of alpha in position i
+        for i in range(self.L):
+            for alpha in range(self.num_symbols):
+                self.f_ij[i, i, alpha, alpha] = self.f_i[i, alpha]
+
+        # compute pseudo counted frequencies
+        # from raw frequencies
+        self.regularize_f_i()
+        self.regularize_f_ij()
+
+    def encode_unused_fields(self):
+        """
+        Set plmc-specific parameters
+        (lambda_J, lambda_group, num_iter)
+        to an arbitrary numerical value.
+
+        Note:
+        The field lambda_h is used to store
+        the pseudo count (the negative sign
+        simply serves as marker).
+        """
+        self.lambda_J = -1
+        self.lambda_group = -1
+        self.num_iter = -1
+        self.lambda_h = -self.pseudo_count
+
+    def decode_unused_fields(self, save_pseudo_count=True):
+        """
+        Set plmc-specific parameters
+        to None to ensure correct usage
+        of the object.
+
+        Note:
+        If save_pseudo_count is True,
+        the pseudo count (that was temporarily
+        stored in lambda_h) is stored in the
+        appropriate field.
+
+        Parameters
+        ----------
+        save_pseudo_count : bool, optional (default: True)
+            If True, the pseudo count is read from
+            the field lambda_h and stored in the
+            appropriate field.
+        """
+        self.lambda_J = None
+        self.lambda_group = None
+        self.num_iter = None
+
+        if save_pseudo_count:
+            self.pseudo_count = -self.lambda_h
+
+        self.lambda_h = None
+
+    def to_file(self, out_file, precision="float32", file_format="plmc_v1"):
```

Comment:
Yeah, I know.. The thing is that a mean-field model right now can only be read in in plmc_v2 format (the whole adaption process is only called in __read_plmc_v2(). So, writing to a plmc_v1 model doesn't really make sense - so, I wanted to prevent that from happening. (I had to specify the file_format parameter though for backward compatibility). Anyway, I agree that this is not nice. What do you think? Still want me to pass the file format to the super-call?